components/merino/src/merino.rs

```diff
@@ -0,0 +1,844 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+use serde::{Deserialize, Serialize};
+use url::Url;
+use uuid::Uuid;
+
+// Locales supported by Merino Curated Reccomendations
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum Locale {
+    #[serde(rename = "fr")]
+    Fr,
+    #[serde(rename = "fr-FR")]
+    FrFr,
+    #[serde(rename = "es")]
+    Es,
+    #[serde(rename = "es-ES")]
+    EsEs,
+    #[serde(rename = "it")]
+    It,
+    #[serde(rename = "it-IT")]
+    ItIt,
+    #[serde(rename = "en")]
+    En,
+    #[serde(rename = "en-CA")]
+    EnCa,
+    #[serde(rename = "en-GB")]
+    EnGb,
+    #[serde(rename = "en-US")]
+    EnUs,
+    #[serde(rename = "de")]
+    De,
+    #[serde(rename = "de-DE")]
+    DeDe,
+    #[serde(rename = "de-AT")]
+    DeAt,
+    #[serde(rename = "de-CH")]
+    DeCh,
+}
+// Configuration settings for a Section
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
```

Comment:
gut feeling told me that i should be adding something like `uniffi::Record` but it wasn't too happy with me....

Reply:
It should just work. What errors are you getting?

Reply:
Maybe you need `uniffi::setup_scaffolding!()` in `lib.rs`?